### PR TITLE
+str Add Sink.never

### PR DESCRIPTION
**Pull request by He-Pin**
References https://github.com/akka/akka/issues/24896

### CODE REVIEW

File: akka-stream/src/main/scala/akka/stream/impl/Stages.scala [hunk 1]
```diff
@@ -135,6 +136,7 @@ import akka.stream.Attributes._
     val publisherSink = name("publisherSink")
     val fanoutPublisherSink = name("fanoutPublisherSink")
     val ignoreSink = name("ignoreSink")
+    val backpressureSink = name("backpressureSink") and inputBufferZero
```
Comment (hepin1989) [inline akka-stream/src/main/scala/akka/stream/impl/Stages.scala:139]: I think a always backpressure sink do not need an operator buffer.

**Review by johanandren — APPROVED** [review]
LGTM

File: akka-stream/src/main/scala/akka/stream/javadsl/Sink.scala [hunk 2]
```diff
@@ -85,6 +85,12 @@ object Sink {
   def ignore[T](): Sink[T, CompletionStage[Done]] =
     new Sink(scaladsl.Sink.ignore.toCompletionStage())
 
+  /**
+   * A [[Sink]] that will always backpressure and never consume any elements from the stream.
+   * */
+  def backpressure[T]: Sink[T, CompletionStage[Done]] =
```
Comment (patriknw) [inline akka-stream/src/main/scala/akka/stream/javadsl/Sink.scala:91]: I think the name `backpressure` may cause some confusion. Looking at corresponding `Source` we have `Source.never`. Could `Sink.never` be a better name?

Comment (hepin1989) [reply akka-stream/src/main/scala/akka/stream/javadsl/Sink.scala:91]: Cool.

Comment (hepin1989) [reply akka-stream/src/main/scala/akka/stream/javadsl/Sink.scala]: @patriknw I have updated it with `never`

File: akka-stream/src/main/scala/akka/stream/scaladsl/Sink.scala [hunk 3]
```diff
@@ -301,6 +301,12 @@ object Sink {
    */
   def ignore: Sink[Any, Future[Done]] = fromGraph(GraphStages.IgnoreSink)
 
+  /**
+   * A [[Sink]] that will always backpressure never cancel and never consume any elements from the stream.
+   * */
+  def never: Sink[Any, Future[Done]] = _never
+  private[this] val _never: Sink[Any, Future[Done]] = fromGraph(GraphStages.NeverSink)
```
[the remark this answers is not in the document]
Comment (hepin1989) [reply akka-stream/src/main/scala/akka/stream/scaladsl/Sink.scala:308]: private[this] will be dropped in scala3

[the remark this answers is not in the document]
Comment (johanandren) [reply akka-stream/src/main/scala/akka/stream/scaladsl/Sink.scala:308]: Scala 3 build passing though, so it is still allowed?

**Review by johanandren — APPROVED** [review]

File: akka-stream/src/main/scala/akka/stream/scaladsl/Sink.scala (hunk 3 above)
[the remark this answers is not in the document]
Comment (hepin1989) [reply akka-stream/src/main/scala/akka/stream/scaladsl/Sink.scala:308]: it's still valid for now.

**Review by patriknw — APPROVED** [review]
LGTM

**Pull request merged**